process/block/preprocess/rewardTxPreProcessor.go

```diff
@@ -208,8 +209,9 @@ func (rtp *rewardTxPreprocessor) RestoreBlockDataIntoPools(
 
 // ProcessBlockTransactions processes all the reward transactions from the block.Body, updates the state
 func (rtp *rewardTxPreprocessor) ProcessBlockTransactions(
-	_ data.HeaderHandler,
+	headerHandler data.HeaderHandler,
 	body *block.Body,
+	processedMiniBlocks *processedMb.ProcessedMiniBlockTracker,
```

Comment:
If the pointer does not change between the subsequent block processing operations, this is a valid point

Reply:
As I explained above this will be a subject of a bigger refactor task which was added in Jira -> EN-12158 with a detailed description and all the steps which should be done